server/src/test/java/org/apache/calcite/test/ServerTest.java

```diff
@@ -127,6 +127,16 @@ static Connection connect() throws SQLException {
         assertThat(r.getInt(1), is(1));
         assertThat(r.next(), is(false));
       }
+
+      b = s.execute("create schema if not exists s");
+      assertThat(b, is(false));
```

Comment:
Why not true?
Schema not created successfully?

Reply:
Vladimir, would you please add `reason` messages to the asserts so it is clear why given values are expected?

Reply:
@xy2953396112 `false` means that the result is not a `ResultSet`

Reply:
@vlsi I do not see any value in it, since these asserts check not for the outcome of the DDL operation per see, but whether JDBC returned a `ResultSet` or not. Obviously, for all DDL statements, it would be `false`. The incorrect behavior is checked through an exception that would be thrown otherwise. E.g., if `CREATE OR REPLACE` schema does not re-create the schema, it would not be possible to create a table with the same name as the table that existed in the previous shcema.

Reply:
> I do not see any value in it, since these asserts check not for the outcome of the DDL operation per se, but whether JDBC command returned a ResultSet or not

Frankly speaking, that is not really obvious from the code, even though I admit `statement.execute` is the very basic JDBC API.

I would suggest:
1) If the outcome of `execute` is not needed, then skip the assert. Just ignore the returned value.
2) If you want to assert something, use `reason` method or add a helper method like `assertExecutesWithoutResutlSet(s, "create schema if not exists s")`. Then the test code would be shorter, it would be easier to understand, and the failure messages could be way more readable. For instance, the helper method could throw errors like "statement produced resultset with columns ... while no resultset was expected".

Reply:
I agree with these comments, but the goal of the ticket was to fix particular bugs, not refactor the test. The introduced changes do not make the test more confusing, since they follow the established approach which is used dozens of times in the same test class. We may refactor the test if needed in a separate issue.

Reply:
> The introduced changes do not make the test more confusing

The introduced changes make it hard to understand the nature of the changes.
I do not ask you to refactor all the existing asserts in the test.
I just kindly ask you to use a better style for the newly added code.

The question from xy2953396112 highlights that it is not really clear what `true/false` means in this context (it is tempting to assume that `true/false` means success/failure which turns out to be wrong).

Thanks.

Reply:
Makes sense. I removed these assertions completely, as they are unrelated to the fix. Also added comments to the tests.

Reply:
Comments in code get out of date extremely fast, so it is better to put them into the exception message.
What do you think of `assertDoesNotThrow` with a relevant `reason` message?

Reply:
Makes sense, I update the tests to use `assertDoesNotThrow`.